pyanaconda/ui/gui/spokes/source.py

```diff
@@ -418,6 +418,11 @@ def _method_changed(self):
             # Make sure anaconda doesn't touch this device.
             part.protected = True
             self.storage.config.protectedDevSpecs.append(part.name)
+            if part.type == "partition" and part.isLogical:
+                # mark the extended partition as protected too
+                extended = self.storage.devicetree.getDeviceByPath(part.disk.format.extendedPartition.path)
+                extended.protected = True
+                self.storage.config.protectedDevSpecs.append(extended.name)
```

Comment:
I think this should check for the name in the ``protectedDevSpecs`` list because if there are multiple such logical partitions, we don't to add the logical partition's name multiple times.